lib/turn.rb

```diff
@@ -54,34 +54,59 @@ def row_win
 
   def diagonal_rightup
     #these represent the six arrays that a right_up diagonal win can occur in
-    arr1 = [board.columns[0].rows[3], board.columns[1].rows[2], board.columns[2].rows[1], board.columns[3].rows[0]]
-    arr2 = [board.columns[0].rows[4], board.columns[1].rows[3], board.columns[2].rows[2], board.columns[3].rows[1], board.columns[4].rows[0]]
-    arr3 = [board.columns[0].rows[5], board.columns[1].rows[4], board.columns[2].rows[3], board.columns[3].rows[2], board.columns[4].rows[1], board.columns[5].rows[0]]
-    arr4 = [board.columns[1].rows[5], board.columns[2].rows[4], board.columns[3].rows[3], board.columns[4].rows[2], board.columns[5].rows[1], board.columns[6].rows[0]]
-    arr5 = [board.columns[2].rows[5], board.columns[3].rows[4], board.columns[4].rows[3], board.columns[5].rows[2], board.columns[6].rows[1]]
-    arr6 = [board.columns[3].rows[5], board.columns[4].rows[4], board.columns[5].rows[3], board.columns[6].rows[2]]
-
-    right_up = [arr1, arr2, arr3, arr4, arr5, arr6]
-
+    arr0 = [board.columns[0].rows[3].player, board.columns[1].rows[2].player, board.columns[2].rows[1].player, board.columns[3].rows[0].player]
+    arr1 = [board.columns[0].rows[4].player, board.columns[1].rows[3].player, board.columns[2].rows[2].player, board.columns[3].rows[1].player, board.columns[4].rows[0].player]
+    arr2 = [board.columns[0].rows[5].player, board.columns[1].rows[4].player, board.columns[2].rows[3].player, board.columns[3].rows[2].player, board.columns[4].rows[1].player, board.columns[5].rows[0].player]
+    arr3 = [board.columns[1].rows[5].player, board.columns[2].rows[4].player, board.columns[3].rows[3].player, board.columns[4].rows[2].player, board.columns[5].rows[1].player, board.columns[6].rows[0].player]
+    arr4 = [board.columns[2].rows[5].player, board.columns[3].rows[4].player, board.columns[4].rows[3].player, board.columns[5].rows[2].player, board.columns[6].rows[1].player]
+    arr5 = [board.columns[3].rows[5].player, board.columns[4].rows[4].player, board.columns[5].rows[3].player, board.columns[6].rows[2].player]
+    right_up = [arr0, arr1, arr2, arr3, arr4, arr5]
+    
+    #key checks are in arr0[0], arr1[1], arr2[2], arr3[2], arr4[1], arr5[0]
```

Comment:
I like how you put a comment on the key checks I mentioned in my earlier comment here! Helps with following the code!

Reply:
I'm working on documenting more frequently!